From 8dcc601ca437d7f86531d2ca655798a7a73972c6 Mon Sep 17 00:00:00 2001
From: Roger Monk <r-monk@ti.com>
Date: Mon, 13 Aug 2012 21:44:49 +0100
Subject: [PATCH 086/102] board-am335xevm: Add Beaglebone Motor Cape Support

   * Beaglebone Motor Cape adds 4 DRV8833 devices connected to
     the 8 PWM outputs on the beaglebone allowing 8 DC Motors or
     4 Stepper Motors to be controlled
      * If PWM channels are not available some of these channels can be
        remapped to GPIO pins via jumpers

   * GPIOs are used for Direction control of each motor channel (1 per channel)
   * GPIOs are also used for Sense Feedback from each motor (2 per channel)
      * Channel 1 sense pins are mapped to QEP0 for use with quadrature encoders

   * Initial Patch recognises the Motor Cape and makes all necessary
     pins available by ensuring w1 and spi1 are marked as in-use

Signed-off-by: Roger Monk <r-monk@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 699fab3..f2d4a69 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -3165,6 +3165,12 @@ static void beaglebone_cape_setup(struct memory_accessor *mem_acc, void *context
 		cssp_gpmc_init();
 	}
 
+	if (!strncmp("BB-BONE-MOTOR-01", cape_config.partnumber, 16)) {
+		pr_info("BeagleBone cape: initializing Motor cape\n");
+		beaglebone_w1gpio_free = 0;
+		beaglebone_spi1_free = 0;
+        }
+
 	goto out2;
 out:
 	/*
-- 
1.8.1.4

